### PR TITLE
doc, chainparams: 29775 release notes and follow-ups

### PR DESCRIPTION
**Pull request by fjahr**
This completes follow-ups left open in #29775.

- Adds release notes
- Addresses the [misalignment](https://github.com/bitcoin/bitcoin/pull/29775#discussion_r1706982102) in deprecation warnings and hints at the intention to remove support for Testnet3.
- Adds initial minimum chainwork for Testnet4.
- Use the `MAX_TIMEWARP` constant as the timewarp defense delta, equal to `MAX_FUTURE_BLOCK_TIME`.

### CODE REVIEW

File: src/chainparamsbase.cpp [hunk 1]
```diff
@@ -17,7 +17,7 @@ void SetupChainParamsBaseOptions(ArgsManager& argsman)
     argsman.AddArg("-regtest", "Enter regression test mode, which uses a special chain in which blocks can be solved instantly. "
                  "This is intended for regression testing tools and app development. Equivalent to -chain=regtest.", ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::CHAINPARAMS);
     argsman.AddArg("-testactivationheight=name@height.", "Set the activation height of 'name' (segwit, bip34, dersig, cltv, csv). (regtest-only)", ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::DEBUG_TEST);
-    argsman.AddArg("-testnet", "Use the testnet3 chain. Equivalent to -chain=test. Support for testnet3 is deprecated and will be removed with the next release. Consider moving to testnet4 now by using -testnet4.", ArgsManager::ALLOW_ANY, OptionsCategory::CHAINPARAMS);
+    argsman.AddArg("-testnet", "Use the testnet3 chain. Equivalent to -chain=test. Support for testnet3 is deprecated and will be removed in an upcoming release (current target v30). Consider moving to testnet4 now by using -testnet4.", ArgsManager::ALLOW_ANY, OptionsCategory::CHAINPARAMS);
```
Comment (tdb3) [inline src/chainparamsbase.cpp:20]: I'm partial to omitting a specific release number (e.g. and maybe including it just in release notes instead), but It's not something I feel super strongly about.

File: src/init.cpp [hunk 2]
```diff
@@ -917,7 +917,7 @@ bool AppInitParameterInteraction(const ArgsManager& args)
 
     // Testnet3 deprecation warning
     if (chain == ChainType::TESTNET) {
-        LogInfo("Warning: Support for testnet3 is deprecated and will be removed in an upcoming release. Consider switching to testnet4.\n");
+        LogInfo("Warning: Support for testnet3 is deprecated and will be removed in an upcoming release (current target v30). Consider switching to testnet4.\n");
```
Comment (tdb3) [inline src/init.cpp:920]: Looks like `feature_config_args.py` needs to be updated to reflect the updated log.  Something like:

```diff
diff --git a/test/functional/feature_config_args.py b/test/functional/feature_config_args.py
index bb20e2baa85..201112ba0d5 100755
--- a/test/functional/feature_config_args.py
+++ b/test/functional/feature_config_args.py
@@ -378,7 +378,7 @@ class ConfArgsTest(BitcoinTestFramework):
 
     def test_testnet3_deprecation_msg(self):
         self.log.info("Test testnet3 deprecation warning")
-        t3_warning_log = "Warning: Support for testnet3 is deprecated and will be removed in an upcoming release. Consider switching to testnet4."
+        t3_warning_log = "Warning: Support for testnet3 is deprecated and will be removed in an upcoming release (current target v30). Consider switching to testnet4."
 
         def warning_msg(node, approx_size):
             return f'Warning: Disk space for "{node.datadir_path / node.chain / "blocks" }" may not accommodate the block files. Approximately {approx_size} GB of data will be stored in this directory.'
```

**Review by tdb3 — COMMENTED** [review]
Concept ACK

File: src/chainparamsbase.cpp (hunk 1 above)
Comment (fjahr) [reply src/chainparamsbase.cpp]: Sigh, I guess several people feel this way, so I have removed it now.

File: src/init.cpp (hunk 2 above)
Comment (fjahr) [reply src/init.cpp]: Automatically fixed since I have removed the version reference.

**Comment by fjahr** [comment]
Removed reference to a specific version in the deprecation warning and addressed [@Sjors comment](https://github.com/bitcoin/bitcoin/pull/29775#issuecomment-2275425772) by using the `MAX_FUTURE_BLOCK_TIME` explicitly as the timewarp defense delta.

**Review by tdb3 — COMMENTED** [review]
Approach ACK
The commit message of f2f3d802c9dbc096808dca8924501ca90081d813 can be updated to remove "and mention v30 target"

**Comment by fjahr** [comment]
> The commit message of [f2f3d80](https://github.com/bitcoin/bitcoin/commit/f2f3d802c9dbc096808dca8924501ca90081d813) can be updated to remove "and mention v30 target"

Fixed

**Review by tdb3 — APPROVED** [review]
ACK 35ad61a6bde14a74cc07dc8ecf3bb1cd49590e46

File: src/validation.cpp [hunk 3]
```diff
@@ -4188,7 +4188,7 @@ static bool ContextualCheckBlockHeader(const CBlockHeader& block, BlockValidatio
         // Check timestamp for the first block of each difficulty adjustment
         // interval, except the genesis block.
         if (nHeight % consensusParams.DifficultyAdjustmentInterval() == 0) {
-            if (block.GetBlockTime() < pindexPrev->GetBlockTime() - 60 * 60 * 2) {
+            if (block.GetBlockTime() < pindexPrev->GetBlockTime() - MAX_FUTURE_BLOCK_TIME) {
```
Comment (Sjors) [inline src/validation.cpp:4191]: 35ad61a6bde14a74cc07dc8ecf3bb1cd49590e46: let's define a separate constant and make it clear these are different but related:

```cpp
/**
 * Maximum number of seconds that the timestamp of the first
 * block of a difficulty adjustment period is allowed to
 * be earlier than the last block of the previous period.
 */
static constexpr int64_t MAX_TIMEWARP{MAX_FUTURE_BLOCK_TIME};

/**
 * If the timestamp of the last block in a difficulty period is set
 * MAX_FUTURE_BLOCK_TIME seconds in the future, an honest miner should
 * be able to mine the first block using the current time.
 */
static_assert(MAX_FUTURE_BLOCK_TIME <= MAX_TIMEWARP);
```

See ongoing discussion in https://github.com/bitcoin/bips/pull/1658

Comment (Sjors) [reply src/validation.cpp:4191]: And see #30614.

File: src/kernel/chainparams.cpp [hunk 4]
```diff
@@ -334,7 +334,7 @@ class CTestNet4Params : public CChainParams {
         consensus.vDeployments[Consensus::DEPLOYMENT_TAPROOT].nTimeout = Consensus::BIP9Deployment::NO_TIMEOUT;
         consensus.vDeployments[Consensus::DEPLOYMENT_TAPROOT].min_activation_height = 0; // No activation delay
 
-        consensus.nMinimumChainWork = uint256{};
+        consensus.nMinimumChainWork = uint256{"000000000000000000000000000000000000000000000056faca98a0cd9bdf5f"};
```
Comment (Sjors) [inline src/kernel/chainparams.cpp:337]: 1163b08378a50a9be00ced434d55f1b04bc9dea6: I get the same value

**Review by Sjors — COMMENTED** [review]
ACK 35ad61a6bde14a74cc07dc8ecf3bb1cd49590e46 modulo the above comment

File: src/validation.cpp (hunk 3 above)
Comment (fjahr) [reply src/validation.cpp]: > let's define a separate constant

done

**Comment by fjahr** [comment]
Addressed @Sjors feedback and introduced a separate constant `MAX_TIMEWARP`.

File: src/validation.cpp (hunk 3 above)
Comment (fjahr) [reply src/validation.cpp]: > And see https://github.com/bitcoin/bitcoin/issues/30614, but that should probably be its own followup.

Agree, it seems that we currently have the safer option given we don't really know what other mining software is doing (at least that's my feeling). If we do still get consensus on tightening the limit we can do that in another follow-up in the next two weeks but it shouldn't hold up this PR here.

**Review by tdb3 — APPROVED** [review]
re ACK 92c1d7d1f8664fe2d259cc609c87f603609b2b60
The new constant adds clarity.

**Comment by Sjors** [comment]
ACK 92c1d7d1f8664fe2d259cc609c87f603609b2b60

**Comment by achow101** [comment]
ACK 92c1d7d1f8664fe2d259cc609c87f603609b2b60

**Pull request merged**